client/src/components/drawer/index.jsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { MDCTemporaryDrawer } from '@material/drawer/dist/mdc.drawer.min.js';
+
+export default class Drawer extends React.PureComponent {
+  static get propTypes() {
+    return {
+      children: React.PropTypes.oneOfType([
+        React.PropTypes.element,
+        React.PropTypes.arrayOf(React.PropTypes.oneOfType([React.PropTypes.element, React.PropTypes.arrayOf(React.PropTypes.element)]))
+      ]),
+      header: React.PropTypes.string,
+      onClick: React.PropTypes.func,
+      onCloseRequest: React.PropTypes.func,
+      open: React.PropTypes.bool.isRequired
+    };
+  }
+
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.handleDrawerClick = this.handleDrawerClick.bind(this);
+    this.handleEscape = this.handleEscape.bind(this);
+  }
+
+  componentDidMount() {
+    this.mdcDrawer = new MDCTemporaryDrawer(this.root);
+    this.mdcDrawer.open = this.props.open;
+    this.root.addEventListener('click', this.handleClick);
+    this.drawer.addEventListener('click', this.handleDrawerClick);
+    window.addEventListener('keyup', this.handleEscape);
+  }
+
+  componentDidUpdate() {
+    if (this.mdcDrawer) {
+      this.mdcDrawer.open = this.props.open;
+    }
+  }
+
+  componentWillUnmount() {
+    this.root.removeEventListener('click', this.handleClick);
+    this.drawer.removeEventListener('click', this.handleDrawerClick);
+    window.removeEventListener('keyup', this.handleEscape);
+  }
+
+  handleClick(e) {
+    if (this.props.onCloseRequest) {
+      this.props.onCloseRequest(e);
+    }
+  }
+
+  handleDrawerClick(e) {
+    if (this.props.onClick) {
+      this.props.onClick(e);
+    }
+  }
+
+  handleEscape(e) {
+    if (e.code === 'Escape' && this.props.onCloseRequest) {
```

Comment:
It's supported in Chrome, Firefox, and (most importantly) Safari. [link](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code)